identity-core/src/crypto/signature/signature.rs

```diff
@@ -100,19 +138,47 @@ impl Serialize for Signature {
   {
     let hide: bool = self.__hide();
 
-    let mut state: S::SerializeMap = if hide {
-      serializer.serialize_map(Some(1 + 5))?
+    // TODO: why was this + 5??
```

Comment:
I could also not figure this out.

Reply:
Went back in time, turns out the `Signature` used to have its own `SignatureOptions` when it was first created, with five extra fields! So it's most likely a holdover from that time. Also pretty interesting that a very similar `SignatureOptions` was originally present then removed in 19e813a016973f166ba3fbf59f5c709f7fdb7a64, but they used to be tied to the verification methods themselves rather than general properties.

https://github.com/iotaledger/identity.rs/blob/199da87b20febbda1ea6277899fdd4178a4f953d/identity-did/src/signature/signature.rs

```rust
impl Debug for Signature {
  fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
    f.debug_struct("Signature")
      .field("type_", &self.type_)
      .field("data", &self.data)
      .field("verification_method", &self.options.verification_method)
      .field("proof_purpose", &self.options.proof_purpose)
      .field("created", &self.options.created)
      .field("nonce", &self.options.nonce)
      .field("domain", &self.options.domain)
      .finish()
  }
}
```

Conclusion: safe to remove.